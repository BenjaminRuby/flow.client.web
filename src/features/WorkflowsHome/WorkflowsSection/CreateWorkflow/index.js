import React, { Component } from "react";
import PropTypes from "prop-types";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { actions as workflowActions } from "State/workflow";
import { actions as workflowRevisionActions } from "State/workflowRevision";
import ModalWrapper from "@boomerang/boomerang-components/lib/Modal";
import ModalFlow from "@boomerang/boomerang-components/lib/ModalFlow";
import { notify, Notification } from "@boomerang/boomerang-components/lib/Notifications";
import Tooltip from "@boomerang/boomerang-components/lib/Tooltip";
import Button from "@boomerang/boomerang-components/lib/Button";
import { Add32 } from "@carbon/icons-react";
import CreateWorkflowContent from "./CreateWorkflowContent";
import { BASE_SERVICE_URL } from "Config/servicesConfig";
import DiagramApplication, { createWorkflowRevisionBody } from "Utilities/DiagramApplication";

export class CreateWorkflow extends Component {
  static propTypes = {
    actions: PropTypes.object.isRequired,
    workflowRevisionActions: PropTypes.object.isRequired,
    fetchTeams: PropTypes.func.isRequired,
    team: PropTypes.object.isRequired
  };

  diagramApp = new DiagramApplication({ dag: null, isLocked: false });

  createWorkflow = workflowData => {
    const { workflowActions, workflowRevisionActions, fetchTeams } = this.props;
    let workflowId;
    return workflowActions
      .create(`${BASE_SERVICE_URL}/workflow`, workflowData)
      .then(res => {
        workflowId = res.data.id;
        const dagProps = createWorkflowRevisionBody(this.diagramApp, "Create workflow");
        const workflowRevision = {
          ...dagProps,
          workflowId
        };
        workflowActions.setHasUnsavedWorkflowUpdates({ hasUpdates: false });
        fetchTeams();
        return workflowRevisionActions.create(`${BASE_SERVICE_URL}/workflow/${workflowId}/revision`, workflowRevision);
      })
      .then(res => {
        notify(
          <Notification type="success" title="Create Workflow" message="Successfully created workflow and version" />
        );
        this.props.history.push(`/editor/${workflowId}/designer`);
      })
      .catch(err => {
        notify(<Notification type="error" title="Something's wrong" message="Failed to create workflow and version" />);
        return Promise.reject();
      });
  };
  render() {
    const { team, teams, isCreating } = this.props;

    return (
      <ModalWrapper
        ModalTrigger={() => (
          <Button className="b-workflow-placeholder">
            <div className="b-workflow-placeholder__box">
              <Add32 data-tip data-for={team.id} className="b-workflow-placeholder__plus" />
            </div>
            <Tooltip id={team.id}>Create Workflow</Tooltip>
          </Button>
        )}
        modalContent={(closeModal, rest) => (
          <ModalFlow
            headerTitle="Create a new Workflow"
            headerSubtitle="Get started with these basics, then proceed to designing it out."
            closeModal={closeModal}
            createWorkflow={this.createWorkflow}
            confirmModalProps={{ affirmativeAction: closeModal }}
            team={team}
            teams={teams}
            isCreating={isCreating}
            theme="bmrg-white"
            {...rest}
          >
            <CreateWorkflowContent />
          </ModalFlow>
        )}
      />
    );
  }
}

const mapStateToProps = state => {
  return {
    teams: state.teams.data,
    isCreating: state.workflow.isCreating
  };
};

const mapDispatchToProps = dispatch => {
  return {
    workflowActions: bindActionCreators(workflowActions, dispatch),
    workflowRevisionActions: bindActionCreators(workflowRevisionActions, dispatch)
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CreateWorkflow);
