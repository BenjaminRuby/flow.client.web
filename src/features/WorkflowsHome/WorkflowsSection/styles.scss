@import "~@boomerang/boomerang-styles/lib/colors";

.c-workflow-section {
  display: flex;
  flex-direction: column;
  width: 100%;
  height: auto;
  margin-top: 3rem;
  .bmrg--b-button {
    margin-right: 1rem;
  }

  .bmrg--b-tooltip.--bmrg-flow {
    border: 0rem;
    margin-top: 0.625rem;
  }
}
.c-workflow-section__header {
  display: flex;
  justify-content: space-between;
  align-items: center;
}
.b-workflow-section__team {
  margin-left: 1rem;
  font-size: 1.25rem;
  font-weight: 500;
  margin: 0rem;
  letter-spacing: 0.0625rem;
}
.b-workflow-section__button {
  margin-right: 1rem;
}
.c-workflow-section__workflows {
  display: flex;
  flex-wrap: wrap;
}

.b-workflow-placeholder {
  border: none;
  background: none;
  cursor: pointer;
  margin: 1rem 0.75rem;
  padding: 0rem;
}

.b-workflow-placeholder__box {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 22rem;
  height: 9.325rem;
  border-radius: 0.25rem;
  border: 0.0625rem dashed $cool-gray-20;
  position: relative;
}

.b-workflow-placeholder__text {
  font-size: 3.5rem;
  line-height: 1rem;
  font-weight: 200;
  color: $cool-gray-50;
}
