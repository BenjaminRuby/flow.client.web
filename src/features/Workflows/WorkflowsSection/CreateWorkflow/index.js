import React, { Component } from "react";
import PropTypes from "prop-types";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { actions as workflowActions } from "State/workflow";
import { actions as workflowRevisionActions } from "State/workflowRevision";
import { ModalFlow, notify, ToastNotification } from "@boomerang/carbon-addons-boomerang-react";
import { Add32 } from "@carbon/icons-react";
import CreateWorkflowContent from "./CreateWorkflowContent";
import { BASE_SERVICE_URL } from "Config/servicesConfig";
import DiagramApplication, { createWorkflowRevisionBody } from "Utilities/DiagramApplication";
import styles from "./createWorkflow.module.scss";

export class CreateWorkflow extends Component {
  static propTypes = {
    fetchTeams: PropTypes.func.isRequired,
    team: PropTypes.object.isRequired,
    workflowActions: PropTypes.object.isRequired,
    workflowRevisionActions: PropTypes.object.isRequired
  };

  diagramApp = new DiagramApplication({ dag: null, isLocked: false });

  createWorkflow = workflowData => {
    const { workflowActions, workflowRevisionActions, fetchTeams } = this.props;
    let workflowId;
    return workflowActions
      .create(`${BASE_SERVICE_URL}/workflow`, workflowData)
      .then(res => {
        workflowId = res.data.id;
        const dagProps = createWorkflowRevisionBody(this.diagramApp, "Create workflow");
        const workflowRevision = {
          ...dagProps,
          workflowId
        };
        fetchTeams();
        return workflowRevisionActions.create(`${BASE_SERVICE_URL}/workflow/${workflowId}/revision`, workflowRevision);
      })
      .then(res => {
        notify(
          <ToastNotification
            kind="success"
            title="Create Workflow"
            subtitle="Successfully created workflow and version"
          />
        );
        this.props.history.push(`/editor/${workflowId}/designer`);
      })
      .catch(err => {
        notify(
          <ToastNotification kind="error" title="Something's wrong" subtitle="Failed to create workflow and version" />
        );
        return Promise.reject();
      });
  };
  render() {
    const { team, teams, isCreating } = this.props;

    return (
      <ModalFlow
        modalTrigger={({ openModal }) => (
          <button id="create-wokflow-card" className={styles.container} onClick={openModal}>
            <Add32 className={styles.addIcon} />
            <label htmlFor="create-wokflow" className={styles.text}>
              Create a new workflow
            </label>
          </button>
        )}
        confirmModalProps={{
          title: "Close this?",
          children: "Your request will not be saved"
        }}
        modalHeaderProps={{
          title: "Create a new Workflow",
          subtitle: "Get started with these basics, then proceed to designing it out."
        }}
      >
        <CreateWorkflowContent createWorkflow={this.createWorkflow} team={team} teams={teams} isCreating={isCreating} />
      </ModalFlow>
    );
  }
}

const mapStateToProps = state => {
  return {
    teams: state.teams.data,
    isCreating: state.workflow.isCreating
  };
};

const mapDispatchToProps = dispatch => {
  return {
    workflowActions: bindActionCreators(workflowActions, dispatch),
    workflowRevisionActions: bindActionCreators(workflowRevisionActions, dispatch)
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CreateWorkflow);
