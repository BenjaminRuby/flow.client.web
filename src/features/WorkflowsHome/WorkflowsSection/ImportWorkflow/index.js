import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { actions as importWorkflowActions } from "State/importWorkflow";
import Modal from "@boomerang/boomerang-components/lib/Modal";
import ModalFlow from "@boomerang/boomerang-components/lib/ModalFlow";
import { notify, Notification } from "@boomerang/boomerang-components/lib/Notifications";
import { Icon } from "carbon-components-react";
import ImportAttachment from "./ImportAttachment";
import ImportConfirm from "./ImportConfirm";
import ImportType from "./ImportType";
import { BASE_SERVICE_URL } from "Config/servicesConfig";
import "./styles.scss";

class ImportWorkflow extends Component {
  static propTypes = {
    fetchTeams: PropTypes.func.isRequired,
    importWorkflowActions: PropTypes.object.isRequired,
    importWorkflowState: PropTypes.object.isRequired
  };

  handleImportWorkflow = (data, isUpdate, closeModal) => {
    return this.props.importWorkflowActions
      .post(`${BASE_SERVICE_URL}/workflow/import?update=${isUpdate}`, JSON.parse(data))
      .then(() => {
        notify(
          <Notification
            type="success"
            title={` ${isUpdate ? "Update" : "Import"} Workflow`}
            message={`Workflow successfully ${isUpdate ? "updated" : "imported"}`}
          />
        );
        closeModal();
        this.props.fetchTeams();
      })
      .catch(err => {
        //noop
      });
  };

  render() {
    const initialState = {
      step: 0,
      formData: {
        files: [],
        isUpdate: false
      }
    };

    return (
      <Modal
        modalProps={{ shouldCloseOnOverlayClick: false }}
        ModalTrigger={() => (
          <button>
            <Icon
              data-tip
              data-for={this.props.teamId}
              className="b-workflow-import__icon"
              name="icon--upload"
              alt="Import Workflow"
            />
          </button>
        )}
        initialState={initialState}
        theme="bmrg-flow"
        modalContent={(closeModal, rest) => (
          <ModalFlow
            headerTitle="Workflow Import"
            headerSubtitle="Import your own workflow"
            closeModal={closeModal}
            confirmModalProps={{ affirmativeAction: closeModal, theme: "bmrg-flow" }}
            theme="bmrg-flow"
            {...rest}
          >
            <ImportType />
            <ImportAttachment />
            <ImportConfirm
              fetchTeams={this.props.fetchTeams}
              handleImportWorkflow={this.handleImportWorkflow}
              importWorkflowActions={this.props.importWorkflowActions}
              importWorkflowState={this.props.importWorkflowState}
            />
          </ModalFlow>
        )}
      />
    );
  }
}

const mapStateToProps = state => ({
  importWorkflowState: state.importWorkflow
});

const mapDispatchToProps = dispatch => ({
  importWorkflowActions: bindActionCreators(importWorkflowActions, dispatch)
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ImportWorkflow);
