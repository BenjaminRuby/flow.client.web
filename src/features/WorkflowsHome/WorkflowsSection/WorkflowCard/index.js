import React, { Component } from "react";
import PropTypes from "prop-types";
import { withRouter } from "react-router-dom";
import { OverflowMenu, OverflowMenuItem } from "carbon-components-react";
import Tooltip from "@boomerang/boomerang-components/lib/Tooltip";
import AlertModal from "@boomerang/boomerang-components/lib/AlertModal";
import ConfirmModal from "@boomerang/boomerang-components/lib/ConfirmModal";
import Modal from "@boomerang/boomerang-components/lib/Modal";
import ModalFlow from "@boomerang/boomerang-components/lib/ModalFlow";
import WorkflowInputModalContent from "./WorkflowInputModalContent";
import imgs from "Assets/icons";
import "./styles.scss";
import playButton from "./img/playButton.svg";

class WorkflowCard extends Component {
  static propTypes = {
    history: PropTypes.object.isRequired,
    workflow: PropTypes.object.isRequired,
    updateWorkflows: PropTypes.func.isRequired,
    executeWorkflows: PropTypes.func.isRequired
  };

  executeWorkflow = (redirect, inputProps) => {
    this.props.executeWorkflow({ workflowId: this.props.workflow.id, redirect, inputProps });
  };

  render() {
    const { workflow, history, teamId, deleteWorkflow } = this.props;
    const menuOptions = [
      {
        itemText: "Edit",
        onClick: () => history.push(`/editor/${workflow.id}/designer`),
        primaryFocus: false
      },
      {
        itemText: "Activity",
        onClick: () => history.push(`/activity/${workflow.id}`),
        primaryFocus: false
      },
      {
        itemText: "Delete",
        primaryFocus: false,
        isDelete: true
      }
    ];

    return (
      <div className="c-workflow-card">
        <div className="c-workflow-card__header">
          <div className="c-workflow-card__status">
            <div className={`b-workflow-card__circle --${workflow.status}`} />
            <label className="b-workflow-card__status">{workflow.status}</label>
          </div>
          <OverflowMenu className="b-workflow-card__menu" ariaLabel="" iconDescription="">
            {menuOptions.map(option => {
              return option.isDelete ? (
                <AlertModal
                  ModalTrigger={() => (
                    <OverflowMenuItem
                      className="b-workflow-card__option"
                      requireTitle={false}
                      onClick={option.onClick}
                      itemText={option.itemText}
                      primaryFocus={option.primaryFocus}
                      key={option.itemText}
                    />
                  )}
                  modalContent={(closeModal, rest) => (
                    <ConfirmModal
                      closeModal={closeModal}
                      affirmativeAction={() => {
                        deleteWorkflow({ teamId, workflowId: workflow.id });
                      }}
                      title="DELETE THIS WORKFLOW?"
                      subTitleTop="It will be gone. Forever."
                      negativeText="NO"
                      affirmativeText="DELETE"
                      theme="bmrg-white"
                    />
                  )}
                />
              ) : (
                <OverflowMenuItem
                  className="b-workflow-card__option"
                  requireTitle={false}
                  onClick={option.onClick}
                  itemText={option.itemText}
                  primaryFocus={option.primaryFocus}
                  key={option.itemText}
                />
              );
            })}
          </OverflowMenu>
        </div>
        <div className="c-workflow-card__info">
          <div className="c-workflow-card__icon">
            <img className="b-workflow-card__icon" src={imgs[workflow.icon ? workflow.icon : "docs"]} alt="icon" />
          </div>
          <div className="c-workflow-card__description">
            <h2 className="b-workflow-card__name">{workflow.name}</h2>
            <p className="b-workflow-card__description">{workflow.shortDescription}</p>
            <span data-tip data-for={workflow.id} className="b-workflow-card-launch">
              {workflow.inputs && workflow.inputs.length > 0 ? (
                <Modal
                  ModalTrigger={() => (
                    <img src={playButton} className="b-workflow-card-launch__icon" alt="Execute workflow" />
                  )}
                  modalContent={(closeModal, rest) => (
                    <ModalFlow
                      headerTitle="Workflow Inputs"
                      headerSubtitle="Supply some values"
                      components={[{ step: 0, component: WorkflowInputModalContent }]}
                      closeModal={closeModal}
                      confirmModalProps={{ affirmativeAction: closeModal }}
                      inputs={workflow.inputs}
                      executeWorkflow={this.executeWorkflow}
                      theme="bmrg-white"
                      {...rest}
                    />
                  )}
                />
              ) : (
                <AlertModal
                  className="bmrg--c-alert-modal --execute-workflow"
                  ModalTrigger={() => (
                    <img src={playButton} className="b-workflow-card-launch__icon" alt="Execute workflow" />
                  )}
                  modalContent={closeModal => (
                    <ConfirmModal
                      style={{ width: "32rem", height: "28rem" }}
                      title="Execute workflow?"
                      subTitleTop="It will run"
                      closeModal={closeModal}
                      affirmativeAction={() => this.executeWorkflow(false)}
                      affirmativeText="Run"
                      theme="bmrg-white"
                    >
                      <button
                        className="bmrg--b-confirm-modal__button --affirmative --children"
                        onClick={() => this.executeWorkflow(true)}
                      >
                        Run and View
                      </button>
                    </ConfirmModal>
                  )}
                />
              )}
            </span>
            <Tooltip id={workflow.id} place="bottom" theme="bmrg-white">
              Execute workflow
            </Tooltip>
          </div>
        </div>
      </div>
    );
  }
}

export default withRouter(WorkflowCard);
