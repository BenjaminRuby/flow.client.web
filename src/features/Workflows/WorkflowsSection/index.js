import React, { Component } from "react";
import PropTypes from "prop-types";
import { withRouter } from "react-router-dom";
import CreateWorkflow from "./CreateWorkflow";
import ImportWorkflow from "./ImportWorkflow";
import WorkflowCard from "./WorkflowCard";
import styles from "./workflowsSection.module.scss";

class WorkflowSection extends Component {
  static propTypes = {
    deleteWorkflow: PropTypes.func.isRequired,
    executeWorkflow: PropTypes.func.isRequired,
    fetchTeams: PropTypes.func.isRequired,
    searchQuery: PropTypes.string.isRequired,
    setActiveTeam: PropTypes.func.isRequired,
    setActiveTeamAndRedirect: PropTypes.func.isRequired,
    team: PropTypes.object.isRequired
  };

  render() {
    const {
      deleteWorkflow,
      executeWorkflow,
      fetchTeams,
      history,
      setActiveTeam,
      setActiveTeamAndRedirect,
      searchQuery,
      team
    } = this.props;

    let workflows = [];
    if (searchQuery) {
      workflows = team.workflows.filter(workflow => workflow.name.toLowerCase().includes(searchQuery.toLowerCase()));
    } else {
      workflows = team.workflows;
    }

    return (
      <section className={styles.container}>
        <div className={styles.header}>
          <h1 className={styles.team}>{`${team.name} (${workflows.length})`}</h1>
          <ImportWorkflow fetchTeams={fetchTeams} teamId={team.id} />
        </div>
        <div className={styles.workflows}>
          {workflows.map(workflow => (
            <WorkflowCard
              deleteWorkflow={deleteWorkflow}
              executeWorkflow={executeWorkflow}
              key={workflow.id}
              setActiveTeam={setActiveTeam}
              teamId={team.id}
              workflow={workflow}
            />
          ))}
          <CreateWorkflow
            team={team}
            setActiveTeamAndRedirect={setActiveTeamAndRedirect}
            history={history}
            fetchTeams={fetchTeams}
          />
        </div>
      </section>
    );
  }
}

export default withRouter(WorkflowSection);
