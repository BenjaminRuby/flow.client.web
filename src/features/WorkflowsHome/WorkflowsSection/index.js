import React, { Component } from "react";
import PropTypes from "prop-types";
import { withRouter } from "react-router-dom";
import Button from "@boomerang/boomerang-components/lib/Button";
import Tooltip from "@boomerang/boomerang-components/lib/Tooltip";
import ImportWorkflow from "./ImportWorkflow";
import WorkflowCard from "./WorkflowCard";
import "./styles.scss";

class WorkflowSection extends Component {
  static propTypes = {
    team: PropTypes.object.isRequired,
    importWorkflow: PropTypes.object.isRequired,
    searchQuery: PropTypes.string.isRequired,
    setActiveTeamAndRedirect: PropTypes.func.isRequired,
    fetchTeams: PropTypes.func.isRequired,
    executeWorkflow: PropTypes.func.isRequired,
    deleteWorkflow: PropTypes.func.isRequired,
    handleImportWorkflow: PropTypes.func.isRequired,
    setActiveTeam: PropTypes.func.isRequired
  };

  render() {
    const {
      deleteWorkflow,
      executeWorkflow,
      fetchTeams,
      setActiveTeam,
      setActiveTeamAndRedirect,
      searchQuery,
      team
    } = this.props;

    let workflows = [];
    if (searchQuery) {
      workflows = team.workflows.filter(workflow => workflow.name.toLowerCase().includes(searchQuery.toLowerCase()));
    } else {
      workflows = team.workflows;
    }

    return (
      <section className="c-workflow-section">
        <div className="c-workflow-section__header">
          <h1 className="b-workflow-section__team">{team.name}</h1>
          <ImportWorkflow fetchTeams={fetchTeams} teamId={team.name} />
          <Tooltip id={team.name}>Import Workflow</Tooltip>
        </div>
        <div className="c-workflow-section__workflows">
          {workflows.map(workflow => (
            <WorkflowCard
              deleteWorkflow={deleteWorkflow}
              executeWorkflow={executeWorkflow}
              key={workflow.id}
              setActiveTeam={setActiveTeam}
              teamId={team.id}
              workflow={workflow}
            />
          ))}
          <Button className="b-workflow-placeholder" onClick={() => setActiveTeamAndRedirect(team.id)}>
            <div className="b-workflow-placeholder__box">
              <div data-tip data-for={team.id} className="b-workflow-placeholder__text">
                +
              </div>
            </div>
            <Tooltip id={team.id}>Create Workflow</Tooltip>
          </Button>
        </div>
      </section>
    );
  }
}

export default withRouter(WorkflowSection);
