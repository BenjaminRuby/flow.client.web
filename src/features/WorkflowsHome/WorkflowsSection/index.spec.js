import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router";
import WorkflowsSection from "./index";

const mockfn = jest.fn();
jest.mock("@boomerang/boomerang-components", () => ({
  Button: "Button",
  Tooltip: "Tooltip"
}));

const searchQuery = "test";
const setActiveTeamAndRedirect = mockfn;
const updateWorkflows = mockfn;
const setActiveTeam = mockfn;
const executeWorkflow = mockfn;
const deleteWorkflow = mockfn;
const fetchTeams = mockfn;
const handleImportWorkflow = mockfn;
const closeModal = mockfn;
const team = {
  id: "1234",
  name: "Lucas' team",
  workflows: [
    {
      id: "hQDkX9v",
      name: "lucas-workflow-1",
      description: "blablabla",
      status: "published",
      icon: "utility"
    },
    {
      id: "456",
      name: "lucas-workflow-2",
      description: "blablabla",
      status: "published",
      icon: "utility"
    },
    {
      id: "789",
      name: "lucas-workflow-3",
      description: "blablabla",
      status: "draft",
      icon: "secure"
    }
  ]
};
const history = {};

describe("WorkflowsSection --- Snapshot", () => {
  it("Capturing Snapshot of WorkflowsSection", () => {
    const { baseElement } = global.rtlReduxRouterRender(
      <MemoryRouter>
        <WorkflowsSection
          history={history}
          searchQuery={searchQuery}
          team={team}
          updateWorkflows={updateWorkflows}
          executeWorkflow={executeWorkflow}
          deleteWorkflow={deleteWorkflow}
          setActiveTeam={setActiveTeam}
          setActiveTeamAndRedirect={setActiveTeamAndRedirect}
          importWorkflow={{}}
          fetchTeams={fetchTeams}
          handleImportWorkflow={handleImportWorkflow}
          closeModal={closeModal}
        />
      </MemoryRouter>
    );
    expect(baseElement).toMatchSnapshot();
  });
});
