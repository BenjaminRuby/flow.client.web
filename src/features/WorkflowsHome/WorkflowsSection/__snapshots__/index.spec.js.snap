// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkflowsSection --- Snapshot Capturing Snapshot of WorkflowsSection 1`] = `
<body>
  <div>
    <section
      class="c-workflow-section"
    >
      <div
        class="c-workflow-section__header"
      >
        <h1
          class="b-workflow-section__team"
        >
          Lucas' team
        </h1>
        <button>
          <svg
            alt="Import Workflow"
            aria-hidden="true"
            class="b-workflow-import__icon"
            currentItem="false"
            data-for="1234"
            data-tip="true"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            style="will-change: transform;"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 9l.7.7 3.8-3.8V15h1V5.9l3.8 3.8.7-.7-5-5zm0-5V2h10v2h1V2c0-.6-.4-1-1-1H3c-.6 0-1 .4-1 1v2h1z"
            />
          </svg>
        </button>
        <div
          class="bx--modal bx--bmrg-modal-flow"
          role="presentation"
          tabindex="-1"
        >
          <div
            class="bx--modal-container bx--bmrg-modal-flow-container"
          >
            <div
              class="bx--modal-header"
            >
              <p
                class="bx--modal-header__heading bx--type-beta"
              >
                Workflow Import
              </p>
              <p
                class="bx--bmrg-modal-flow-subtitle"
              >
                Import your own workflow
              </p>
              <button
                class="bx--modal-close"
                title="Close the modal"
                type="button"
              >
                <svg
                  aria-label="Close the modal"
                  class="bx--modal-close__icon"
                  focusable="false"
                  height="20"
                  preserveAspectRatio="xMidYMid meet"
                  role="img"
                  style="will-change: transform;"
                  viewBox="0 0 32 32"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4l6.6 6.6L8 22.6 9.4 24l6.6-6.6 6.6 6.6 1.4-1.4-6.6-6.6L24 9.4z"
                  />
                </svg>
              </button>
            </div>
            <ul
              class="bx--progress"
            >
              <li
                class="bx--progress-step bx--progress-step--current"
              >
                <div
                  class="bx--progress-step-button bx--progress-step-button--unclickable"
                  role="button"
                  tabindex="-1"
                >
                  <svg>
                    <path
                      d="M 7, 7 m -7, 0 a 7,7 0 1,0 14,0 a 7,7 0 1,0 -14,0"
                    />
                    <title />
                  </svg>
                  <p
                    class="bx--progress-label"
                  >
                    Type
                  </p>
                  <span
                    class="bx--progress-line"
                  />
                </div>
              </li>
              <li
                class="bx--progress-step bx--progress-step--incomplete"
              >
                <div
                  class="bx--progress-step-button bx--progress-step-button--unclickable"
                  role="button"
                  tabindex="-1"
                >
                  <svg>
                    <title />
                    <path
                      d="M8 1C4.1 1 1 4.1 1 8s3.1 7 7 7 7-3.1 7-7-3.1-7-7-7zm0 13c-3.3 0-6-2.7-6-6s2.7-6 6-6 6 2.7 6 6-2.7 6-6 6z"
                    />
                  </svg>
                  <p
                    class="bx--progress-label"
                  >
                    Attachment
                  </p>
                  <span
                    class="bx--progress-line"
                  />
                </div>
              </li>
              <li
                class="bx--progress-step bx--progress-step--incomplete"
              >
                <div
                  class="bx--progress-step-button bx--progress-step-button--unclickable"
                  role="button"
                  tabindex="-1"
                >
                  <svg>
                    <title />
                    <path
                      d="M8 1C4.1 1 1 4.1 1 8s3.1 7 7 7 7-3.1 7-7-3.1-7-7-7zm0 13c-3.3 0-6-2.7-6-6s2.7-6 6-6 6 2.7 6 6-2.7 6-6 6z"
                    />
                  </svg>
                  <p
                    class="bx--progress-label"
                  >
                    Confirm
                  </p>
                  <span
                    class="bx--progress-line"
                  />
                </div>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="__react_component_tooltip place-top type-dark "
          data-id="tooltip"
          id="Lucas' team"
        >
          Import Workflow
        </div>
      </div>
      <div
        class="c-workflow-section__workflows"
      >
        <button
          class="b-workflow-placeholder"
        >
          <div
            class="b-workflow-placeholder__box"
          >
            <svg
              aria-hidden="true"
              class="b-workflow-placeholder__plus"
              currentItem="false"
              data-for="1234"
              data-tip="true"
              focusable="false"
              height="32"
              preserveAspectRatio="xMidYMid meet"
              style="will-change: transform;"
              viewBox="0 0 32 32"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17 15V7h-2v8H7v2h8v8h2v-8h8v-2h-8z"
              />
            </svg>
          </div>
          <div
            class="__react_component_tooltip place-top type-dark "
            data-id="tooltip"
            id="1234"
          >
            Create Workflow
          </div>
        </button>
      </div>
    </section>
  </div>
</body>
`;
